tripy/tests/flat_ir/ops/test_gather.py

```diff
@@ -16,28 +16,37 @@
 #
 
 import tripy as tp
-from tripy import int32
 
 from tripy.flat_ir.ops import DynamicGatherOp
 from tripy.frontend.trace import Trace
-
+import pytest
 
 class TestGatherOp:
-    def test_gather_str(self):
-        data = tp.Tensor([3.0, 4.0], name="data")
-        index = tp.Tensor([0], dtype=int32, name="indices")
-        out = tp.gather(data, 0, index)
+    @pytest.mark.parametrize("axis", [0,1,2])
+    def test_gather_str_dim0(self, axis):
+        data = tp.Tensor([[[0.0000, 0.0000],
+                           [0.0000, 0.0000],
+                           [0.0000, 0.0000]],
+
+                          [[1.0000, 1.0000],
+                           [1.0000, 1.0000],
+                           [1.0000, 1.0000]],
+
+                          [[2.0000, 2.0000],
+                           [2.0000, 2.0000],
+                           [2.0000, 2.0000]]], name="data")
+        index = tp.Tensor([1], dtype=tp.int32, name="indices")
+        out = tp.gather(data, axis, index)
         out.name = "out"
 
         trace = Trace([out])
         flat_ir = trace.to_flat_ir()
 
         gather = flat_ir.ops[-1]
         reshape = flat_ir.ops[-2]
-
         print(str(reshape))
         assert isinstance(gather, DynamicGatherOp)
         assert (
             str(gather)
-            == "out: [rank=(1), dtype=(float32), loc=(gpu:0)] = DynamicGatherOp(data, indices, t_inter3, axis=0)"
+            == f"out: [rank=(3), dtype=(float32), loc=(gpu:0)] = DynamicGatherOp(data, indices, t_inter3, axis={axis})"
```

Comment:
I think the more important thing to test is the FlatIR->MLIR translation. Can you add a test for that? You can look at some of the other files in this directory for examples.

Reply:
See `tripy/tests/flat_ir/ops/test_iota.py`

Reply:
I added some test cases, but I am not sure how much of the MLIR should be checked.

Reply:
Oh my mistake, looks like the fix for non-zero axis was actually in the trace op. Could you add a test to check the trace->FlatIR translation? We should make sure we're generating the right set of ops. @parthchadha could probably provide more details on what exactly we should look out for

Reply:
Oh my mistake, looks like the fix for non-zero axis was actually in the trace op. Could you add a test to check the trace->FlatIR translation? We should make sure we're generating the right set of ops. @parthchadha could probably provide more details on what exactly we should look out for. The PR description says it fixes the case where `axis > 0`, but the code seems like it's for the case where `axis == rank - 1`. Not entirely sure I've understood what's happening there.

Reply:
Which PR are you referring to? And I thought the test case that I wrote already checks the translation from trace->FlatIR.

Reply:
This commit: https://github.com/NVIDIA/TensorRT-Incubator/commit/c0784f34844f65600f95038555f44c2b52b1cfd1
It does implicitly test trace->FlatIR, but I think we should explicitly test it also in `tripy/tests/frontend/trace/ops/test_gather.py`

Reply:
The old was completely broken when axis > 0 (slicing on the shape tensor was incorrect for `slice_sizes`)

Reply:
Added more trace->FlatIR testing for when axis>0.